students/MicahBraun/Lesson 04/Markov_Analysis.py

```diff
@@ -0,0 +1,100 @@
+    except KeyError:
+        potential_words_map[root_word] = [word]            # if key of root_word does not already exist in dict, create
+
+    root_word = arrange_trigram(root_word, word)          # call arrange_trigram for root_word, word
+
+
+def randomize(n=300):
+    """Generates random words from the analyzed text -- n: number of words to generate"""
+    global big_list                                                 # accesses global var big_list
+    start = random.choice(list(potential_words_map.keys()))         # a random value (word) from dict
+
+    for i in range(n):                                              # for items in range 0-300
+        potential_words = potential_words_map.get(start, None)      # potential_words = random
+        if potential_words == None:                                 # if == None, randomize(300 - 300)
+            randomize(n-i)                                          # randomize all
+            return                                              # Not None
+        word = random.choice(potential_words)                       # word = random selection from potential_words
+        big_list.append(word + " ")                                 # append word to big_list along with whitespace
+        start = arrange_trigram(start, word)                       # start = call arrange_trigram() on start, word
+
+
+def arrange_trigram(t, word):
+    """Creates a new tuple with t (start) as the tuple and word as the end added on to the end (the
+    beginning of the new tuple has been cut off ( see [1:] ) """
+    return t[1:] + (word,)                                          # returns new tuple
+
+
+def list_to_str_format():
+    """Imports and uses regular expressions to break up long string of text and display
+    in multiple lines for better viewing of trigrams"""
+
+    global big_list                                                 # access global big_list
+    whole = "".join(str(i) for i in big_list)   # whole = join each element in big_list (this is why I added whitespace)
+    regex = re.compile('(.*?(\s))*')            # regex = re.compile((any character != \n\t\r) (repeat <-- 0:n times
+                                                # (repeat 0-1 times for resulting -->)(any whitespace)(repeat 0:n)
+    while whole != "":                                              # while whole isn't empty string:
+        break_pos = regex.match(whole[:100]).end()                  # break_pos breaks after 100 words on whitespace
+        print(whole[:break_pos])                                    # print whole up to break_pos
+        whole = whole[break_pos:]                                   # whole = whole starting from previous break_pos
+
+
+def main(filename, n=300, order=3):                                 # main function calls all other functions
+    n = int(n)                                                      # n = default length of text to read-in from file
```

Comment:
The typical pattern in Python would be to specify type of the input parameter in a docstring